src/drivers/common/memory.c

```diff
@@ -100,4 +101,16 @@ static int periph_memory_init(void) {
 	return 0;
 }
 
+int periph_desc(struct periph_memory_desc **buff) {
+	struct periph_memory_desc *elem;
+
+	int i = 0;
+	array_spread_foreach(elem, __periph_mem_registry) {
+		buff[i] = elem;
+		i++;
```

Comment:
By the way, you can just write `buf[i++] = elem;`. That is, `i` will be post-incremented immediately after assignment.